Tasks/ServiceFabricDeploy/ServiceFabricSDK/Publish-UpgradedServiceFabricApplication.ps1

```diff
@@ -218,67 +218,69 @@ function Publish-UpgradedServiceFabricApplication
         $reg = Get-ServiceFabricApplicationType -ApplicationTypeName $names.ApplicationTypeName | Where-Object  { $_.ApplicationTypeVersion -eq $names.ApplicationTypeVersion }
         if ($reg)
         {
-            Write-Host (Get-VstsLocString -Key SFSDK_UnregisteringExistingAppType -ArgumentList @($names.ApplicationTypeName, $names.ApplicationTypeVersion))
-            $reg | Unregister-ServiceFabricApplicationType -Force
```

Comment:
This is a breaking behavior. Should we upgrade task major version? Or we can add another task input which asks user whether he wants to use existing registered application type package or wants to un-register existing package and always use new package.
I am fine with bumping up major version - for two reasons:

- If old application package with same version exists, most probably it is fine to use this package. Only in an edge case user would have uploaded and registered with a particular version and then wants to register a new package with same version

- Not in favor of adding more inputs unless necessary. we already have a handful. 

@NCarlsonMSFT  what do you think about this?

Reply:
I vote bump the major version

Reply:
I had discussion with @sachinma and @RoopeshNair regarding this. We are tweaking the functionality a bit. Apart from existing check that an application package with same type and version exists, we will additionally check **whether any application in cluster is using this particular application type and version**. If **yes**,then we will use the existing application package for upgrade[**New behavior**]. If **no**, then we will unregister old package and register new package and use that for upgrade[**Old behavior**]. 
Since, we are not breaking old behavior, we only need to bump task's minor version. 
@kuvinodms, please update the PR with this change.
@NCarlsonMSFT, does this sound good to you?